lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -4092,6 +4180,93 @@ class AdjointEmitter final : public SILInstructionVisitor<AdjointEmitter> {
+          cotangentVectorTy->getStructOrBoundGenericStruct();
+      assert(cotangentVectorDecl);
+      // Find the corresponding field in the cotangent space.
+      VarDecl *correspondingField = nullptr;
+      // If the cotangent space is the original struct, then field is the same.
+      if (cotangentVectorDecl == seai->getStructDecl())
+        correspondingField = seai->getField();
+      // Otherwise, look up the field by name.
+      else {
+        auto correspondingFieldLookup =
+        cotangentVectorDecl->lookupDirect(seai->getField()->getName());
+        assert(correspondingFieldLookup.size() == 1);
+        correspondingField = cast<VarDecl>(correspondingFieldLookup.front());
+      }
+      // Accumulate adjoint element buffer into adjoint buffer for original
+      // `struct_element_addr` operand.
+      auto adjStructElt =
+          builder.createStructElementAddr(loc,
+              getAdjointBuffer(seai->getOperand()), correspondingField);
+      auto adjElt = getAdjointBuffer(seai);
+      auto *readAccess = builder.createBeginAccess(
+          loc, adjElt, SILAccessKind::Read, SILAccessEnforcement::Static,
+          /*noNestedConflict*/ true, /*fromBuiltin*/ false);
+      addToAdjointBuffer(adjStructElt, readAccess);
+      builder.createEndAccess(loc, readAccess, /*aborted*/ false);
+      return;
+    }
+    case StructExtractDifferentiationStrategy::Getter: {
+      // Construct pullback arguments.
+      // Allocate buffer for pullback indirect result.
+      auto *pullbackResultBuf = builder.createAllocStack(
+          loc, remapType(getCotangentType(seai->getOperand()->getType(),
+                                          getModule())));
+      auto adjElt = getAdjointBuffer(seai);
+
+      // Get the pullback.
+      auto *pullbackField = getPrimalInfo().lookUpPullbackDecl(seai);
+      assert(pullbackField);
+      auto pullback = builder.createStructExtract(
+          loc, primalValueAggregateInAdj, pullbackField);
+
+      // Call the pullback.
+      auto *pullbackCall = builder.createApply(
+          loc, pullback, SubstitutionMap(), {pullbackResultBuf, adjElt},
+          /*isNonThrowing*/ false);
+      assert(pullbackCall->getNumIndirectResults() == 1 &&
+             pullbackCall->getIndirectSILResults().front() ==
+                  pullbackResultBuf);
+
+      // Accumulate pullback result buffer into adjoint buffer for original
+      // `struct_element_addr` operand.
+      auto *readAccess = builder.createBeginAccess(
+          loc, pullbackResultBuf, SILAccessKind::Read,
+          SILAccessEnforcement::Static, /*noNestedConflict*/ true,
+           /*fromBuiltin*/ false);
+      accumulateIndirect(getAdjointBuffer(seai->getOperand()),
+                         pullbackResultBuf);
+      builder.createEndAccess(loc, readAccess, /*aborted*/ false);
+      // Deallocate pullback indirect result buffer.
+      builder.createDeallocStack(loc, pullbackResultBuf);
```

Comment:
```suggestion
      builder.createDeallocStack(loc, pullbackResultBuf);
```
Please release any pullback call's indirect results, since they are immediately accumulated into another buffer.

Reply:
Done in https://github.com/apple/swift/pull/22669/commits/25f16291c2e794966aca760347523b4862acd80b, I believe, by calling `emitCleanup` on `pullbackResultBuf`.
There's an assertion that `pullbackResultBuf` is the only indirect result of the pullback call.

Reply:
Yes, it should be the only result. Getters are simple.